openarm/damiao/encoding.py

```diff
@@ -0,0 +1,731 @@
+
+    Reference: Save parameters response format - custom protocol
+
+    """
+    # Wait for save response with motor's master ID
+    # Timeout to prevent indefinite blocking
+    message = bus.recv(master_id, timeout=0.1)
+
+    # Unpack save response data
+    # Format: '<HBB' = little-endian: slave_id(H) + command(B) + status(B)
+    # slave_id: first 2 bytes, command: byte 3 (0xAA), status: byte 4 (0x01=success)
+    slave_id, command, status = struct.unpack("<HBB", message.data[:4])
+
+    # Check if valid save response (0xAA cmd) and if it succeeded (0x01 status)
+    save_cmd = 0xAA  # Save command identifier
+    success_status = 0x01  # Success status code
+    success = command == save_cmd and status == success_status
+
+    return SaveResponse(
+        slave_id=slave_id,
+        success=success,
+    )
+
+
+def encode_read_register(
+    bus: Bus, slave_id: int, register_address: RegisterAddress
+) -> None:
+    """Read motor register value. Sends CAN message to read register.
+
+    Args:
+        bus: CAN bus instance for message transmission
+        slave_id: Motor slave ID
+        register_address: Register address to read
+
+    Decode with: decode_register_int(bus) or decode_register_float(bus)
+
+    Reference: DM_CAN.py __read_RID_param function lines 381-385
+
+    """
+    # Pack register read command as structured data
+    # Format: '<HBBBBBB' = little-endian: slave_id(H=uint16) + command_bytes(6*B)
+    # Reference: Register read format in DM_CAN.py __read_RID_param function
+    data = struct.pack(
+        "<HBBBBBB",
+        slave_id,  # slave_id as 16-bit value
+        _READ_REGISTER_CODE,  # 0x33 read register command
+        int(register_address),  # register address to read
+        0x00,
+        0x00,
+        0x00,
+        0x00,  # padding bytes
+    )
+
+    # Send to master arbitration ID 0x7FF for register operations
+    # Reference: DM_CAN.py __send_data calls with 0x7FF for register reads
+    message = can.Message(arbitration_id=0x7FF, data=data, is_extended_id=False)
+    bus.send(message)
+
+
+def encode_write_register_int(
+    bus: Bus, slave_id: int, register_address: RegisterAddress, value: int
+) -> None:
+    """Write motor register value as integer. Sends CAN message to write register.
+
+    Args:
+        bus: CAN bus instance for message transmission
+        slave_id: Motor slave ID
+        register_address: Register address to write
+        value: Integer value to write to register
+
+    Decode with: decode_register_int(bus) or decode_register_float(bus)
+
+    Reference: DM_CAN.py __write_motor_param function lines 387-397 (int path)
+
+    """
+    # Pack as uint32 (4 bytes)
+    # Format: '<HBBI' = slave_id(H) + commands(2*B) + value(I=uint32)
+    # Reference: Integer register write format in DM_CAN.py __write_motor_param
+    data = struct.pack(
+        "<HBBI",
+        slave_id,  # slave_id as 16-bit value
+        _WRITE_REGISTER_CODE,  # 0x55 write register command
```

Comment:
i think we can just put `0x55` directly in here:

```suggestion
        0x55,  # write register command
```